bq_table.proto

```diff
@@ -56,4 +56,8 @@ message BigQueryMessageOptions {
   // or "<field name>:RECORD:<protobuf type>" for message types.
   // "NULLABLE" by default, different mode may be set via optional suffix ":<mode>"
   repeated string extra_fields = 3;
+
+  // If set will output the schema file order based on the proto field numbers.
+  // This flag will also apply to all nested message fields.
+  bool sort_by_field_number = 4;
```

Comment:
Thank you for raising the topic, and proposing this change. It generally sounds reasonable, however, here I have a minor concern -- What do you think about having an enum instead of `bool`?

I don't think it is a good idea to that we use field numbers as a "canonical" way to control their order because the field numbers are primarily just unique identifiers of the fields. Protobuf implementations may sort fields by their numbers to make the binary serialization reproducible. However, it is a kind of implementation details.

On the other hand, I understand the use case you explained. It seems acceptable for me to use the field numbers as a "non-canonical" option of the output order. This is why I am proposing `enum`.

Reply:
Hello,

Thanks for the quick feedback! 
I am okay with enum here, but what values did you have in mind for the value set of this enum? 

Would it be a set of potential ways to sort the output big query schema? 

Ex)

// Sort output json schema by
const (
	FIELD_NUMBER string = "field_number"
	ALPHABETICAL       = "alphabetical" 
	....
)

Reply:
The current implementation does not guarantee any order, and I would like to keep it by default. Also, we do not need to support any other orders than your proposing one for now.

Therefore, I would be something like this:

```protobuf
enum FieldOrder {
  FIELD_ORDER_UNSPECIFIED = 0;
  FIELD_ORDER_BY_NUMBER = 1;
}
FieldOrder output_field_order = 4;
```

Reply:
SGTM, I made the required changes which shoudl be in this pull request.